src/webhooks/models/webhookManager.js

```diff
@@ -7,29 +7,49 @@ const webhookDefaultValues = {
     global: false
 };
 
-module.exports.getAllWebhooks = async function () {
+async function getAllWebhooks() {
     let getAllWebhooks = await databaseConnector.getAllWebhooks();
     return getAllWebhooks;
 };
 
-module.exports.getWebhook = async function (webhookId) {
+async function getWebhook(webhookId) {
     const webhook = await databaseConnector.getWebhook(webhookId);
     if (!webhook) {
-        const error = new Error(ERROR_MESSAGES.NOT_FOUND);
-        error.statusCode = 404;
-        throw error;
+        throw generateError(404, ERROR_MESSAGES.NOT_FOUND);
     }
     return webhook;
 };
 
-module.exports.createWebhook = async function(webhookInfo) {
+async function createWebhook(webhookInfo) {
     const webhook = {
         ...webhookDefaultValues,
         ...webhookInfo
     };
     return databaseConnector.createWebhook(webhook);
 };
 
-module.exports.deleteWebhook = async function(webhookId) {
+async function deleteWebhook(webhookId) {
     return databaseConnector.deleteWebhook(webhookId);
-};
\ No newline at end of file
+};
+
+async function updateWebhook(webhookId, webhook) {
+    const webhookInDB = await getWebhook(webhookId);
+    if (!webhookInDB) {
+        throw generateError(404, ERROR_MESSAGES.NOT_FOUND);
+    }
+    return databaseConnector.updateWebhook(webhookId, webhook);
+};
+
+module.exports = {
+    getAllWebhooks,
+    getWebhook,
+    createWebhook,
+    deleteWebhook,
+    updateWebhook
+};
+
+function generateError(stautsCode, message) {
```

Comment:
this is a very generic function that we also have in `processorsManager`, lets find a better place for it

Reply:
done